sdk/program/src/loader_v4.rs

```diff
@@ -41,6 +46,130 @@ impl LoaderV4State {
     }
 }
 
+pub fn is_write_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 0 == instruction_data[0]
+}
+
+pub fn is_truncate_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 1 == instruction_data[0]
+}
+
+pub fn is_deploy_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 2 == instruction_data[0]
+}
+
+pub fn is_retract_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 3 == instruction_data[0]
+}
+
+pub fn is_transfer_authority_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 4 == instruction_data[0]
+}
+
+/// Returns the instructions required to initialize a program/buffer account.
+pub fn create_buffer(
+    payer_address: &Pubkey,
+    buffer_address: &Pubkey,
+    lamports: u64,
+) -> Instruction {
+    system_instruction::create_account(payer_address, buffer_address, lamports, 0, &id())
```

Comment:
This should also take the `size` as a parameter and do the `Truncate` to the given `size` as well.

Reply:
So initial size will be 0, and truncate instruction will set the actual size?

Reply:
Yes, that is the idea. Truncate will also initialize the header struct for the program.